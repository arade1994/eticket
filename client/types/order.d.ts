import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: Date;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
