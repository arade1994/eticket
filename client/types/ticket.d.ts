export interface Ticket {
  id: string;
  title: string;
  price: number;
  userId: string;
  category: string;
  version: number;
}
