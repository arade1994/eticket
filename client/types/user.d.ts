import { type IncomingMessage } from "http";

export interface User {
  id: string;
  firstName: string;
  lastName: string;
  age: number;
  email: string;
  password: string;
}

export interface Rating {
  id: string;
  comment: string;
  rate: number;
  userId: string;
  ratedUserId: string;
}

export interface RequestWithUser extends IncomingMessage {
  currentUser?: { id: string; email: string };
}
