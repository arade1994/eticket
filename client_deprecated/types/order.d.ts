import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: string;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
