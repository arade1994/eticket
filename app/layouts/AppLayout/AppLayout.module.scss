.app {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: var(--color-background);
}

.main {
  flex: 1;
  max-width: 1200px;
  margin: 0 auto;
  padding: 2rem;
  width: 100%;
}
