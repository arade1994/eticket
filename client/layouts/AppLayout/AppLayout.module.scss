.app {
  display: grid;
  grid-template-rows: 50px auto;
  grid-template-columns: 250px 1fr 400px;
  height: 100vh;
  gap: 2px;
  overflow-y: hidden;

  & .contentContainer {
    height: 100%;
    overflow-y: auto;
  }
}
