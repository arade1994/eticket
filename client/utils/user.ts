import { Ticket } from "../types/ticket";
import { User } from "../types/user";

export const getNumOfCreatedTickets = (user: User, tickets: Ticket[]) =>
  tickets?.filter((ticket) => ticket.userId === user.id)?.length;
