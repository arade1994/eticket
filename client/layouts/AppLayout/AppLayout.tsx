import { ElementType } from "react";
import { ToastContainer } from "react-toastify";

import Header from "../../components/Header/Header";
import { Rating, type User } from "../../types/user";

import { Ticket } from "../../types/ticket";

import classes from "./AppLayout.module.scss";
import "react-toastify/dist/ReactToastify.css";
import Navigation from "../../components/Navigation/Navigation";
import NotificationCenter from "../../components/NotificationCenter/NotificationCenter";

interface Props {
  Component: ElementType;
  pageProps: {
    ratings: Rating[];
    tickets: Ticket[];
    users: User[];
  };
  currentUser: User;
}

const AppLayout: React.FC<React.PropsWithChildren<Props>> = ({
  Component,
  pageProps,
  currentUser,
}) => {
  return (
    <div className={classes.app}>
      <Header currentUser={currentUser} />
      {currentUser && <Navigation />}
      <Component currentUser={currentUser} {...pageProps} />
      {currentUser && <NotificationCenter />}
      <ToastContainer
        position="bottom-right"
        autoClose={5000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        theme="colored"
      />
    </div>
  );
};

export default AppLayout;
