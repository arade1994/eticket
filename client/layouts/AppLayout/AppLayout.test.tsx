import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import AppLayout from "./AppLayout";
import mockDb from "../../mock/db.json";
import { Rating, User } from "../../types/user";
import { Ticket } from "../../types/ticket";

const mockRatings = mockDb.ratings as unknown as Rating[];
const mockUsers = mockDb.users as unknown as User[];
const mockTickets = mockDb.tickets as unknown as Ticket[];
const MockComponent: React.FC = () => <div>Component</div>;

describe("<AppLayout />", () => {
  test("it should match a snapshot when rendered by default", () => {
    const { baseElement } = render(
      <AppLayout
        Component={MockComponent}
        pageProps={{
          tickets: mockTickets,
          users: mockUsers,
          ratings: mockRatings,
        }}
        currentUser={mockUsers[1]}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
