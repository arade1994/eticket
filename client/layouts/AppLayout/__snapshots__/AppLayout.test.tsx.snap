// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppLayout /> > it should match a snapshot when rendered by default 1`] = `
<body>
  <div>
    <div
      class="_app_fb33d5"
    >
      <header
        class="_headerContainer_0a265d"
      >
        <nav
          class="_navigation_0a265d"
        >
          <a
            class="_logo_0a265d"
            href="/"
          >
            eTicket
          </a>
          <div
            class="_profileIcon_a7b94c"
          >
            <p
              style="margin: auto;"
            >
              MF
            </p>
          </div>
        </nav>
      </header>
      <aside
        class="_navigationContainer_1558d5"
      >
        <nav
          class="_navigation_1558d5"
        >
          <a
            class=""
            href="/"
          >
            Tickets
          </a>
          <a
            class=""
            href="/orders"
          >
            Orders
          </a>
          <a
            class=""
            href="/users"
          >
            Users
          </a>
          <a
            class="_marginTopAuto_1558d5"
            href="/auth/signout"
          >
            Sign out
          </a>
        </nav>
      </aside>
      <div>
        Component
      </div>
      <aside
        class="_notificationsContainer_2cd674"
      >
        <h1
          style="margin: auto;"
        >
          Notification Center
        </h1>
      </aside>
      <div
        class="Toastify"
      />
    </div>
  </div>
</body>
`;
